### PR TITLE
[99% MODULAR]TESHARI REWORK

### PR DESCRIPTION
**Pull request by TeshariEnjoer**
## About The Pull Request
This PR contains a complete revamp of the Teshari race, with significant tweaks to it in some aspects. Namely the following changes:

- The original Teshari, have been renamed and redesigned into the Raptor race, which will be a replacement for the original. This race is an empty race like the Anthrapomorph race. And they are intended for the beginnings of players. Does not have the standard organizing and features for smaller creatures.
изображение

- The new teshari, in the code represents a "teshari_alternative", fully developed and have a number of distinctive features, including a special temperature mode. The new teshari are extremely vulnerable to any damage, as well as not tolerating high temperatures and fire, but have the ability to good echolocation and mobility.
New Teshari can be picked up by other creatures and in this state be thrown into something or fooled by something.
By the by, teshari have strong limitations in combat and can't throw other creatures, can't perform shove for other creatures, firing heavy weapons will severely affect accuracy. Especially heavy weapons, can do hand damage and knock the shooter down.
изображение

Also, almost all MOD suits and their modules have been redrawn for teshari.
(New sprites)
Screenshot_97
Screenshot_93
(Teshari picked up)
Screenshot_102

## Why It's Good For The Game

This PR, can be good for the game as it adds a lot of new variant wagering, as well as adding full unique features for such a lovely race as the Teshari. At the same time adding missing sprites and sounds.

## Proof Of Testing
изображение

## Changelog

:cl: Fenysha
add: Added two new ablites(agility and echolocation)
add: Added new species Raptor
sound: added teshari sounds(laugh, scream, cough and other)
image: added new sprites for teshari MOD suits, done allmost all
code: Full rework of teshari code
config: added alt teshari to roundstart races
/:cl:

### CODE REVIEW

**Comment by ReturnToZender** [comment]
LET'S FUCKING GOOOOO

Can you work on resprites for broken cosmetics as well?

**Comment by TeshariEnjoer** [comment]
> LET'S FUCKING GOOOOO
> 
> Can you work on resprites for broken cosmetics as well?

In general yes, but I'm not a spriter - I'm specifically a coder. And I only make sprites for the love of raptors.

**Comment by StrangeWeirdKitten** [comment]
> LET'S FUCKING GOOOOO
> 
> 
> 
> Can you work on resprites for broken cosmetics as well?

Out of scope, you can if you want, but I'm not going to require it.

**Comment by StrangeWeirdKitten** [comment]
My biggest thing is that it overrides Teshari itself, which is a widely appreciated species in favor of shifting normal Teshari to be an alt variation. There's 100 percent going to be a drastic switch placed upon multiple characters. I'm mostly curious why it's opted to go the complete replacement route versus a stand alone raptor species people can freely switch into?

Ability wise and species trait wise everything looks pretty solid to hold up as a species.

**Comment by StrangeWeirdKitten** [comment]
Anyway if it gets test merged, it'll be a full commitment. Though I know realistically it won't completely break everything. I think overall in the long term making a Teshari override that makes the base species more open isn't a negative thing. I'll let @Majkl-J do his code review for the more technical stuff. Ability wise, everything is solid.

**Comment by Erol509** [comment]
Looks awesome, pickup mechanics will be really nice, and echolocation sounds interesting. Love it already

**Comment by TeshariEnjoer** [comment]
> My biggest thing is that it overrides Teshari itself, which is a widely appreciated species in favor of shifting normal Teshari to be an alt variation. There's 100 percent going to be a drastic switch placed upon multiple characters and it's not something I can easily test merge. I'm mostly curious why it's opted to go the complete replacement route versus a stand alone raptor species people can freely switch into?
> 
> Ability wise and species trait wise everything looks pretty solid to hold up as a species.
> 
> I don't play Teshari, personally. I have no clue how people would feel or if they strive for a complete replacement. Do they?

Related to this is the fact that many teshari players, do not play teshari themselves - but avali, resomi and the like.

**Comment by TeshariEnjoer** [comment]
I would also add that the transition to the new teshari, is painless. All character owners will have to do is simply change race, after which all changes will be automatically applied.

**Comment by CliffracerX** [comment]
Resprites and bespoke emote sounds are nice QOL that *should* be merged, but leaving the existing negative quirks of being a Teshari on the generic Raptor species & then making actual Teshari significantly mechanically worse in exchange for echolocation and being able to scramble over tables is...not a great look.  Tesh players already eat a lot of mechanical L's, adding more on top isn't going to draw people to the bespoke race even with some shiny new stuff on top.

**Comment by TeshariEnjoer** [comment]
> Resprites and bespoke emote sounds are nice QOL that _should_ be merged, but leaving the existing negative quirks of being a Teshari on the generic Raptor species & then making actual Teshari significantly mechanically worse in exchange for echolocation and being able to scramble over tables is...not a great look. Tesh players already eat a lot of mechanical L's, adding more on top isn't going to draw people to the bespoke race even with some shiny new stuff on top.

In my native language there is such an expression as "Eat the fish and don't get hooked", Teshari is a beautiful race, but I rarely see people that played it in its entirety. Mechanical limitations go side by side with additional abilities. Either way the balance will adjust is a secondary matter.

**Comment by ReturnToZender** [comment]
Not being able to fire heavy weapons is a great flavor trait, the shove trait is probably a little overboard though.

**Comment by TeshariEnjoer** [comment]
> Not being able to fire heavy weapons is a great flavor trait, the shove trait is probably a little overboard though.
> 
> The flashbang downside for Teshari seems to be more than enough balancing for the ability you're giving them
> 
> What is the Echolocation range? 9 Tiles? You can only see for 7 tiles around you, and echolocation should probably not cover the entire screen. I would think something like 3 tiles? Also, it should work only on people that are moving (or only that are running?) that way you can avoid being seen in a locker if you're hiding or similar. There's also a trait that makes your footsteps quieter I believe, echolocation should not be triggered by people w/ that trait

Echolocation in the current state is to compensate for strong weakenings of the teshari. I've had enough playtests to say for sure that this ability will not have a strong impact on balance.

File: modular_zubbers/modules/lpa_teshari/_DEFINES/species_defines.dm
Comment (Majkl-J) [inline modular_zubbers/modules/lpa_teshari/_DEFINES/species_defines.dm:9]: This needs undefines at the end of the folder.

**Review by Majkl-J — CHANGES_REQUESTED** [review]

**Comment by Majkl-J** [comment]
I'm gonna die reviewing this

File: modular_zubbers/modules/lpa_teshari/_DEFINES/species_defines.dm
Comment (TeshariEnjoer) [reply modular_zubbers/modules/lpa_teshari/_DEFINES/species_defines.dm:9]: These defines are used in another file, I can't remove them at the end of the this file.

**Comment by theselfish** [comment]
I agree with Shark here.

If Teshari or whatever snowflake flavour they renamed themselves to don't want the new features, they should be the ones having to swap. The people playing Teshari who want the new content shouldn't be forced to rebuilt their characters.

File: modular_zubbers/modules/lpa_teshari/_DEFINES/species_defines.dm
Comment (Majkl-J) [reply modular_zubbers/modules/lpa_teshari/_DEFINES/species_defines.dm:9]: You can just go to the last file in the folder and undefine it there, cuts on compiler cache sizes

**Comment by Majkl-J** [comment]
> If Teshari or whatever snowflake flavour they renamed themselves to don't want the new features, they should be the ones having to swap. The people playing Teshari who want the new content shouldn't be forced to rebuild their characters.

Shouldn't be an issue with how prefs work, you just swap the race in the options and all parts that can be applied will be applied. We just did the same with abductors and from what I gathered so far there weren't any issues

**Comment by StrangeWeirdKitten** [comment]
Yes what waterpig said

**Comment by Redrover1760** [comment]
Ah, so a reversion to the design old teshari were in on whatever codebases.

Honestly, I've always felt that the old teshari were just too extreme in upsides and downsides. I don't think the species you are playing as should impact how you play this badly, and how others interact with you. Not to mention how powerful extra speed is, but also how if I'm reading the code right any guns that any tesh fires gets inaccuracy by default.

Well, its still not as bad as old tesh, they literally had 50 hp and 1.5x damage multipliers for going mach 7, but regardless. Looking through the code it does seem a bit less absurd. How viable is ranged weaponry, at least light stuff like disablers with tesh? Its not too bad I hope? I'm not sure what heavy weaponry count as, two handed stuff or just sniper rifles and LMGs or stuff? Would basically exclude tesh from stuff like MCRs and ballistic sec then. Maybe fair, considering the speed buff and such. Either way, I find the abilities pretty neat and am personally willing to give this PR a chance as a tesh player.

Also, what does agility do? The PR doesn't exactly state what it does.

**Comment by TeshariEnjoer** [comment]
> Ah, so a reversion to the design old teshari were in on whatever codebases.
> 
> Honestly, I've always felt that the old teshari were just too extreme in upsides and downsides. I don't think the species you are playing as should impact how you play this badly, and how others interact with you. Not to mention how powerful extra speed is, but also how if I'm reading the code right any guns that any tesh fires gets inaccuracy by default.
> 
> Well, its still not as bad as old tesh, they literally had 50 hp and 1.5x damage multipliers for going mach 7, but regardless. Looking through the code it does seem a bit less absurd. How viable is ranged weaponry, at least light stuff like disablers with tesh? Its not too bad I hope? I'm not sure what heavy weaponry count as, two handed stuff or just sniper rifles and LMGs or stuff? Would basically exclude tesh from stuff like MCRs and ballistic sec then. Maybe fair, considering the speed buff and such. Either way, I find the abilities pretty neat and am personally willing to give this PR a chance as a tesh player.
> 
> Also, what does agility do? The PR doesn't exactly state what it does.

The teshari only have restrictions on ranged weapons on heavy weapons. Such as assault rifles, shotguns, sniper rifles and so on. And particularly heavy weapons can even harm them.
In addition, teshari have difficulty throwing and pushing mobs larger than themselves.
Dexterity works quite easily and allows you to instantly climb up on tables and some other obstacles.

**Comment by Redrover1760** [comment]
Also, speaking of the big species changes, any movement speed boost does sorta change the game a ton, and the downsides for it don't seem the most fun to play. In my opinion you could remove it and maybe undo some of the nerfs and it'd probably be for the better. Agility works well as a buff in my opinion, although you could also give tesh another agile-esque thing as well too.

**Comment by TeshariEnjoer** [comment]
> Also, speaking of the big species changes, any movement speed boost does sorta change the game a ton, and the downsides for it don't seem the most fun to play. In my opinion you could remove it and maybe undo some of the nerfs and it'd probably be for the better. Agility works well as a buff in my opinion, although you could also give tesh another agile-esque thing as well too. I'm not really sure how snails are balanced though, now that I think of it.

I've very finely balanced the teshari so that even increasing their speed shouldn't give them any strong advantages. Because all their speed boosts are not levelled by strong weaknesses in combat.

**Comment by Redrover1760** [comment]
> I've very finely balanced the teshari so that even increasing their speed shouldn't give them any strong advantages. Because all their speed boosts are not levelled by strong weaknesses in combat.

This is a bit my point, that since they're nerfed in other areas so hard in order to gain a speed boost sorta forces an identity in combat onto the race. I'm not sure if its going to be for the best or not, but yeah. Also, as a suggestion, you could give em a bonus (or penalty) to tackling, assuming that's still a thing considering tg changed tackling.

**Comment by miniusAreas** [comment]
Crosspost from Discord.
Several major feedback inputs, after watching this PR get looked over by two other maintainers for other servers-

> Teshari Echolocation has a very expensive loop (I don't remember the entirety of what or how), that would run per each teshari, making it a very easy to abuse and possibly create massive time dialation or even lag/crash the server out.

> Teshari and the new species are split because ???, seems like unneeded bloat, something that SS13 suffers from. The species is (almost completely) used by two species, Teshari themselves and Avali- anything else is an edge case and an outlier, and both species are both functionally the same

> Teshari Echolocation is abuseable in a balance sense. This was an issue on other servers that had it before it, and players cannot be trusted to not abuse it in a HRP environment, let alone one like Bubberstation.

> Major grammatical / spelling errors (This can be excused as not everyone is from a primary english-speaking country)

> I (personally) don't consider a compile log as "proof of testing" but that's just a personal gripe.

This PR seems to be (mostly) built upon ideas that I preached in _another server that died and I won't name due to drama reasons._ They were, at one point, tested and merged, and after testing, debate amongst that server and their maintainers, and other factors, we found that Teshari balance like this _**DOES NOT**_ work in a fast-paced, low to medium roleplay environment like TG. This PR would need some work before I would consider it in a reasonable state for testmerge. Codebases like TG do not really support wide and deep changes to species balance with strong pros and cons with how other species work, unlike servers like Baystation where things like this DO work due to their slower pace, and significantly more nuanced mechanics. There's reasons why the difference between most species is very minor things, as very large and diverse pros and cons can make one species much better than the others (in this case, yeah, deffo does that HARD.), and why most species changes don't touch things like movespeed, damage offsets, etc.

**Comment by miniusAreas** [comment]
image


Went back and found the loop in question.

**Comment by DiscordWizard** [comment]
As a Teshari player who plays in security, I just wanna say I can't wait to be an alt teshari with all the downsides while being an officer, it sounds like a fun callenge. 

However the whole unable to shove thing feels a little too harsh considering how the best method to non-lethally detain someone without using half your disabler charge is baton, then shove to make sure they don't stand up, then cuff. Guess that's just gonna be yet more added challenge to the whole thing, I'm still looking forwards to how it will impact my role in security compared to everyone else.

Probably gonna be a lil' scout or criminal catcher, not someone getting into fights.

**Comment by TeshariEnjoer** [comment]
> Crosspost from Discord. Several major feedback inputs, after watching this PR get looked over by two other maintainers for other servers-
> 
> > Teshari Echolocation has a very expensive loop (I don't remember the entirety of what or how), that would run per each teshari, making it a very easy to abuse and possibly create massive time dialation or even lag/crash the server out.
> 
> > Teshari and the new species are split because ???, seems like unneeded bloat, something that SS13 suffers from. The species is (almost completely) used by two species, Teshari themselves and Avali- anything else is an edge case and an outlier, and both species are both functionally the same
> 
> > Teshari Echolocation is abuseable in a balance sense. This was an issue on other servers that had it before it, and players cannot be trusted to not abuse it in a HRP environment, let alone one like Bubberstation.
> 
> > Major grammatical / spelling errors (This can be excused as not everyone is from a primary english-speaking country)
> 
> > I (personally) don't consider a compile log as "proof of testing" but that's just a personal gripe.
> 
> This PR seems to be (mostly) built upon ideas that I preached in _another server that died and I won't name due to drama reasons._ They were, at one point, tested and merged, and after testing, debate amongst that server and their maintainers, and other factors, we found that Teshari balance like this _**DOES NOT**_ work in a fast-paced, low to medium roleplay environment like TG. This PR would need some work before I would consider it in a reasonable state for testmerge. Codebases like TG do not really support wide and deep changes to species balance with strong pros and cons with how other species work, unlike servers like Baystation where things like this DO work due to their slower pace, and significantly more nuanced mechanics. There's reasons why the difference between most species is very minor things, as very large and diverse pros and cons can make one species much better than the others (in this case, yeah, deffo does that HARD.), and why most species changes don't touch things like movespeed, damage offsets, etc.

Let's just say that without serious solutions - you can't get serious results. All teshari developments that were developed by me are completely author's ideas that were formed after long listening to the requests of players, their wishes and comments. 
About teshari server crash - no, it is impossible. I launched more than a hundred teshari with echolocation at once and it did not cause any severe consequences. About speed changes and so on. The balance of new teshari is finely tuned and adjusted to their living conditions.

**Review by Majkl-J — COMMENTED** [review]
Will take a look at this in the next week or so, if you're still around developing this

**Comment by Swiftfeather** [comment]
Hey there, this has been stale for quite a while. If you decide to continue working on it please feel free to re-open this PR!

**Pull request closed without merging**
